packages/wonder-blocks-dropdown/components/action-menu.js

```diff
@@ -20,6 +20,17 @@ type MenuProps = {|
      */
     menuText: string,
 
+    /**
+     * Can be used to override the state of the ActionMenu by parent elements
+     */
+    opened?: boolean,
+
+    /**
+     * In controlled mode, use this prop in case the parent needs to be notified
+     * when the menu opens/closes.
+     */
+    onToggle?: (opened: boolean) => mixed,
```

Comment:
We use `onToggle` on `OptionItem`.  Since the name of the `prop` that controls whether the action menu is opened or closed I think we should use either:
```
onOpenedChanged?: (opened: boolean) => mixed,
```
or
```
onClose?: () => mixed,
```
The latter is more common in our API.  Is there a need to know when `opened` is transitioning from `false => true`?  I wouldn't expect there to be since if the component is controlled then the parent is triggering this transition.

Reply:
I originally used `onClose` but found some issues when trying to completely control this component from its parent. The main reason to change it to `onToggle` was because the trigger element (`Dropdown menu opener`) wasn't aware when the dropdown menu should open. 

`onClose` is good enough for the `Dropdown` component. Because that component works similar to the `ModalLauncher). In Modals, as we are passing the `openModal` as a renderProp, then we let the developer to implement any modal opener from the parent, then they don't need to be aware about when `opened` changes internally.

In `wb-dropdown`,  `Dropdown` works similarly, but the other variants (`ActionMenu`, `SingleSelect` and `MultiSelect`) need to know when to open, as we are hiding the opener implementation details to the devs.

And answering you last question, in dropdowns we know when `opened` transitions from `false` to `true`, and it's when the user clicks on the opener button.

What do you think?

Reply:
That said, we could change the prop name. My intention about naming that prop that way was to use something short that devs could be familiar with and that it could be used (if needed) in another packages. It would be great if we could generalize that behavior to cover more scenarios.

I feel if we restrict that to `onOpenedChange`, then we couldn't reuse the same pattern for other package that don't necessarily use `open/opened` for X reasons.